src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -5889,9 +5889,9 @@ bool Unit::Attack(Unit* victim, bool meleeAttack)
         SetUInt32Value(UNIT_NPC_EMOTESTATE, EMOTE_ONESHOT_NONE);
     }
 
-    // delay offhand weapon attack to next attack time
+    // delay offhand weapon attack by 50% of the base attack time
     if (haveOffhandWeapon() && GetTypeId() != TYPEID_PLAYER)
-        resetAttackTimer(OFF_ATTACK);
+        setAttackTimer(OFF_ATTACK, CalculatePct(GetAttackTime(OFF_ATTACK), 50));
```

Comment:
I am concerned with this - doesn't it have the potential to cause strange (faster than usual) attack patterns when switching targets?

You also seem to assume that `OFF_ATTACK` and `BASE_ATTACK` have the same attack timer here - the code doesn't match the PR description otherwise.

Reply:
they have the same attack timer since offhand attacks of creatures does not rely on weapon attack timers like players do. Instead they use the base attack timer as well. What I did here was only delaying the swing. The timers will still reset after execution.